app/models/hold.rb

```diff
@@ -0,0 +1,22 @@
+# == Schema Information
+#
+# Table name: holds
+#
+#  id               :integer          not null, primary key
+#  thesis_id        :integer          not null
+#  date_requested   :date             not null
+#  date_start       :date             not null
+#  date_end         :date             not null
+#  hold_source_id   :integer          not null
+#  case_number      :string
+#  status           :integer          not null
+#  processing_notes :text
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#
+class Hold < ApplicationRecord
+	belongs_to :thesis
+	belongs_to :hold_source
+
```

Comment:
Adding validations for each required element in the models would be better here and then associated tests to confirm our business rules. The seemingly large amount of tests in this case won't be testing Rails so much as making sure we notice when someone changes the model validations in a way that should break our understanding of the business rules as defined in our tests, we should find out by tests failing.

Right now, I can create a record with no date_end, no date_start, no  date_requested and when I check `.valid?` says true but when I save it it fails with a db constraint error.

```
#<Hold:0x00007fef07641e70
 id: nil,
 thesis_id: 43,
 date_requested: nil,
 date_start: nil,
 date_end: nil,
 hold_source_id: 1,
 case_number: "123456",
 status: "active",
 processing_notes: "Hallo!",
 created_at: Wed, 13 Jan 2021 19:26:44 UTC +00:00,
 updated_at: Wed, 13 Jan 2021 19:26:44 UTC +00:00>
```

☝🏻 that's clearly an invalid Hold by our standards but rails thinks it is valid and will try to save it (and then fail).